diff --git a/node_modules/@nx/js/src/executors/node/node.impl.js b/node_modules/@nx/js/src/executors/node/node.impl.js
index 5614256..66d8836 100644
--- a/node_modules/@nx/js/src/executors/node/node.impl.js
+++ b/node_modules/@nx/js/src/executors/node/node.impl.js
@@ -16,9 +16,25 @@ const get_main_file_dir_1 = require("../../utils/get-main-file-dir");
 const utils_1 = require("nx/src/tasks-runner/utils");
 function debounce(fn, wait) {
     let timeoutId;
+    let pendingPromise = null;
     return () => {
         clearTimeout(timeoutId);
-        timeoutId = setTimeout(fn, wait);
+        if (!pendingPromise) {
+            pendingPromise = new Promise((resolve, reject) => {
+                timeoutId = setTimeout(() => {
+                    fn()
+                        .then((result) => {
+                        pendingPromise = null;
+                        resolve(result);
+                    })
+                        .catch((error) => {
+                        pendingPromise = null;
+                        reject(error);
+                    });
+                }, wait);
+            });
+        }
+        return pendingPromise;
     };
 }
 async function* nodeExecutor(options, context) {
@@ -52,7 +68,7 @@ async function* nodeExecutor(options, context) {
     let additionalExitHandler = null;
     let currentTask = null;
     const tasks = [];
-    yield* (0, async_iterable_1.createAsyncIterable)(async ({ done, next, error }) => {
+    yield* (0, async_iterable_1.createAsyncIterable)(async ({ done, next, error, cleanup }) => {
         const processQueue = async () => {
             if (tasks.length === 0)
                 return;
@@ -163,6 +179,10 @@ async function* nodeExecutor(options, context) {
             await stopAllTasks('SIGHUP');
             process.exit(128 + 1);
         });
+        cleanup(async () => {
+          console.log('running cleanup in the node executor')
+            await stopAllTasks('SIGTERM');
+        });
         if (options.runBuildTargetDependencies) {
             // If a all dependencies need to be rebuild on changes, then register with watcher
             // and run through CLI, otherwise only the current project will rebuild.
